pandas/plotting/_matplotlib/core.py

```diff
@@ -400,9 +402,12 @@ def _compute_plot_data(self):
         # GH16953, _convert is needed as fallback, for ``Series``
         # with ``dtype == object``
         data = data._convert(datetime=True, timedelta=True)
-        numeric_data = data.select_dtypes(
-            include=[np.number, "datetime", "datetimetz", "timedelta"]
-        )
+        select_include_type = [np.number, "datetime", "datetimetz", "timedelta"]
+
+        # GH23719, allow plotting boolean
+        if self.include_bool is True:
```

Comment:
is there a reason we don't just always allow this?

Reply:
https://github.com/pandas-dev/pandas/pull/23813#issuecomment-440255972

Tom used to propose this option to avoid a breaking change for people replying on bool data being dropped @jreback

Reply:
ok

> but for now we can make it False or None if we want to issue a warning and change it to True in the future. 

I think let's makes this default to ``None`` and issue a warning if not explicity passed & there is boolean data. We can change this in future releases.

Reply:
I'm not sure if we necessarily want that. I suspect plotting bools is relatively rare.

Reply:
Hi, my English is not very good, so if i understand correctly, there is no clear outcome of how this should proceed? (`None` with warning or just `False`). I also reckon plotting bools is relatively rare, but i did try to plot at work, and found not doable, so i checked issues and kind of reopen to fix it up. So shall I just merge master to fix failures for now? @TomAugspurger @jreback

Reply:
I'm not sure how to decide which default is more useful, so my preference is to keep things as they are.

Reply:
How about automatically plotting bools (without warning) if there are only bools present (e.g. a 1D boolean series or a DF of all bools)?